File: backend/recapify/src/main/java/com/recapify/llm/LlmService.java
package com.recapify.llm;

import com.recapify.llm.client.MlServiceClient;
import com.recapify.llm.dto.SummaryRequest;
import com.recapify.llm.dto.SummaryResponse;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
@RequiredArgsConstructor
public class LlmService {

    private final MlServiceClient mlServiceClient;

    public SummaryRequest getIntent(String text) {
        return mlServiceClient.extractIntent(text);
    }

    public SummaryResponse getSummary(String text) {
        long startedAt = System.nanoTime();
        SummaryRequest summaryRequest = getIntent(text);
        SummaryResponse summary = mlServiceClient.fetchSummary(summaryRequest);
        long durationMs = (System.nanoTime() - startedAt) / 1_000_000;
        log.info("Summary workflow completed finalSummaryLength={} keyEventsCount={} durationMs={}",
                summary.final_summary().length(), summary.key_events().size(), durationMs);
        return summary;
    }
}

File: frontend/src/app/services/llm.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { EpisodeSummary } from '../models/summary.model';
import { AppLoggerService } from './app-logger.service';
import { tap } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class LlmService {
  private apiUrl = environment.apiUrl;

  constructor(
    private http: HttpClient,
    private logger: AppLoggerService
  ) { }

  getSummary(text: string) {
    const startedAt = performance.now();

    this.logger.info('Requesting episode summary', {
      textLength: text.length
    });

    return this.http.post<EpisodeSummary>(`${this.apiUrl}api/v1/llm/summary`, text).pipe(
      tap({
        next: (summary) => {
          this.logger.info('Episode summary received', {
            durationMs: Math.round(performance.now() - startedAt),
            finalSummaryLength: summary.final_summary.length,
            keyEventsCount: summary.key_events.length
          });
        },
        error: (err) => {
          this.logger.error('Episode summary request failed', {
            durationMs: Math.round(performance.now() - startedAt),
            status: err?.status,
            statusText: err?.statusText,
            url: err?.url
          }, err);
        }
      })
    );
  }
}
